streams/views.py

```diff
@@ -31,4 +32,17 @@ def put(self,request):
                 return Response(status=status.HTTP_400_BAD_REQUEST)
         return Response(status=status.HTTP_401_UNAUTHORIZED) 
 
+ 
+class PostsByStreamView(APIView):
 
+    def get(self,request,pk):
+        user = IsLoggedIn(request)
+        if user is not None:
```

Comment:
I guess there is no need to check whether the user is logged in, posts of a stream should be visible to all. requests